backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts

```diff
@@ -42,21 +59,325 @@ describe("POST /token", () => {
       expect(response.status).toBe(400);
     });
   });
+
+  describe("Given the grant_type value is invalid", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "grant_type=invalid_grant_type",
+        {
+          headers: {
+            "x-custom-auth": "Basic " + toBase64(clientIdAndSecret),
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_grant",
+        error_description: "Invalid grant type or grant type not specified",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given there is no Authorization header in the request", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "grant_type=client_credentials",
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_client",
+        error_description: "Supplied client credentials not recognised",
```

Comment:
I think this one should be a 401?

Reply:
Possibly, that's what the TD says, but the lambda never returned a 401 even though that was in the spec. Should we make the lambda return 401 when the header is missing and 400 when the header is not valid? 😕